linux/src/async_socket_linux.c

```diff
@@ -392,13 +417,9 @@ static void internal_close(ASYNC_SOCKET_HANDLE async_socket)
 
     if (interlocked_add(&async_socket->added_to_completion_port, 0) > 0)
     {
-        completion_port_remove(async_socket->completion_port, async_socket->socket_handle);
+        completion_port_remove(async_socket->completion_port, socket_transport_get_underlying_socket(async_socket->socket_transport_handle));
     }
 
-    // Codes_SRS_ASYNC_SOCKET_LINUX_11_039: [ async_socket_close shall call close on the underlying socket. ]
```

Comment:
>// Codes_SRS_ASYNC_SOCKET_LINUX_11_039: [ async_socket_close shall call close on the underlying socket. ] [](http://example.com/codeflow?start=4&length=105)

Since you removed this one you'll need to remove it everywhere.